app/src/main/java/de/westnordost/streetcomplete/overlays/cycleway/CyclewayOverlay.kt

```diff
@@ -0,0 +1,154 @@
+package de.westnordost.streetcomplete.overlays.cycleway
+
+import de.westnordost.countryboundaries.CountryBoundaries
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.elementfilter.toElementFilterExpression
+import de.westnordost.streetcomplete.data.meta.CountryInfo
+import de.westnordost.streetcomplete.data.meta.CountryInfos
+import de.westnordost.streetcomplete.data.meta.getByLocation
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.filter
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement
+import de.westnordost.streetcomplete.osm.ALL_ROADS
+import de.westnordost.streetcomplete.osm.ANYTHING_UNPAVED
+import de.westnordost.streetcomplete.osm.bicycle_boulevard.BicycleBoulevard
+import de.westnordost.streetcomplete.osm.bicycle_boulevard.createBicycleBoulevard
+import de.westnordost.streetcomplete.osm.cycleway.Cycleway
+import de.westnordost.streetcomplete.osm.cycleway.Cycleway.*
+import de.westnordost.streetcomplete.osm.cycleway.createCyclewaySides
+import de.westnordost.streetcomplete.osm.cycleway.isAmbiguous
+import de.westnordost.streetcomplete.osm.cycleway_separate.SeparateCycleway
+import de.westnordost.streetcomplete.osm.cycleway_separate.createSeparateCycleway
+import de.westnordost.streetcomplete.osm.isPrivateOnFoot
+import de.westnordost.streetcomplete.overlays.Color
+import de.westnordost.streetcomplete.overlays.Overlay
+import de.westnordost.streetcomplete.overlays.PolylineStyle
+import de.westnordost.streetcomplete.overlays.StrokeStyle
+import de.westnordost.streetcomplete.quests.cycleway.AddCycleway
+import java.util.concurrent.FutureTask
+
+class CyclewayOverlay(
+    private val countryInfos: CountryInfos,
+    private val countryBoundaries: FutureTask<CountryBoundaries>
+) : Overlay {
+
+    override val title = R.string.overlay_cycleway
+    override val icon = R.drawable.ic_quest_bicycleway
+    override val changesetComment = "Specify whether there are cycleways"
+    override val wikiLink: String = "Key:cycleway"
+    override val achievements = listOf(EditTypeAchievement.BICYCLIST)
+    override val hidesQuestTypes = setOf(AddCycleway::class.simpleName!!)
+
+    override fun getStyledElements(mapData: MapDataWithGeometry) =
+        // roads
+        mapData.filter("""
+            ways with
+              highway ~ motorway_link|trunk|trunk_link|primary|primary_link|secondary|secondary_link|tertiary|tertiary_link|unclassified|residential|living_street|pedestrian|service
```

Comment:
there are cases of cycleway lanes on motorways

yes really, see https://osmus.slack.com/archives/C029HV951/p1653292353895499

Also, technically `cycleway=track` is still valid there.

Reply:
Hm. I guess it could be treated the same as `motorway_link` and `motorroad`. Do not show as missing, but allow editing.

Though, I wonder which outweighs the other:
- Being able to tag correctly the rare case of a `cycleway` on a motorway (and be it `cycleway=separate`)
- or the addition of wrong cycleway data on motorways due to human errors
?

Reply:
> or the addition of wrong cycleway data on motorways due to human errors

how likely is that user will not spot such mistake?

Also, at least following would apply:

- being able to visualise already tagged info
- being able to remove already badly tagged data on motorways due to human errors

Reply:
Right, okay. I will add motorway